packages/core/src/shared/utilities/cacheUtils.ts

```diff
@@ -10,13 +10,16 @@ import fs from '../../shared/fs/fs'
 import { promises as fsPromises } from 'fs'
 import crypto from 'crypto'
 import { isWeb } from '../extensionGlobals'
+import type { MapSync } from './map'
 
 // TODO(sijaden): further generalize this concept over async references (maybe create a library?)
 // It's pretty clear that this interface (and VSC's `Memento`) reduce down to what is essentially
 // a "pointer" with RW operations that return futures.
 
 /**
  * A general, basic interface for a cache with key associativity.
+ *
+ * Look to use {@link MapSync} instead if you need atomicity.
```

Comment:
what indicates that `KeyedCache` doesn't have atomicity? can be a followup discussion, but... this isn't clear.

Reply:
Making updates requires async read() then write(). So by the time you write the data may have changed due to async. But since MapSync does not use promises the user can read the latest data, then update it atomically.